src/components/Timer/Timer.js

```diff
@@ -1,18 +1,60 @@
-import React, { useState, useEffect } from 'react';
+import React, { useState, useReducer, useEffect } from 'react';
 import useInterval from './../../hooks/useInterval';
+import convertFromSecondToMinutes from './../../utils/convertFromSecondToMinutes';
+import { ACTIONS } from './../../utils/actions';
+
+const initialState = {
+    pomodoro: 20,
+    shortBreak: 5,
+    longBreak: 10,
+};
+
+const intervalOrder = [
+    'pomodoro',
+    'shortBreak',
+    'pomodoro',
+    'shortBreak',
+    'pomodoro',
+    'shortBreak',
+    'pomodoro',
+    'longBreak',
+];
+
+const reducer = (state, action) => {
+    switch (action.type) {
+        case ACTIONS.POMODORO:
+            return { ...state, pomodoro: state.pomodoro - 1 };
```

Comment:
just a note: the way it works now is that we reset the state every time we reach 0 in one of the three timers, so there is not actually a need for `{ ...state, `